src/interface/src/app/plan/plan.component.ts

```diff
@@ -39,22 +36,16 @@ export class PlanComponent implements OnInit, OnDestroy {
     map((show) => (show ? 'SCENARIOS' : 'SCENARIO'))
   );
 
-  scenario$ = this.planStateService.planState$.pipe(
-    switchMap((state) => {
-      if (state.currentScenarioId) {
-        return this.scenarioService.getScenario(state.currentScenarioId);
-      }
-      return of(null);
-    }),
-    catchError((e) => {
-      return of(undefined);
+  scenarioName$ = this.planStateService.planState$.pipe(
+    map((state) => {
+      return state.currentScenarioName || null;
```

Comment:
same here, state.currentScenarioName is already null by default so I dont think you need ` || null`

Reply:
Solved